packages/protocol/contracts/governance/ReleaseGoldInstance.sol

```diff
@@ -0,0 +1,570 @@
+   * @param initialDistributionPercentage Percentage of total rewards available for distribution.
+   *                                      Expressed to 3 significant figures [0, 1000].
+   * @param _canValidate If this schedule's gold can be used for validating.
+   * @param _canVote If this schedule's gold can be used for voting.
+   * @param registryAddress Address of the deployed contracts registry.
+   */
+  constructor(
+    uint256 releaseStartTime,
+    uint256 releaseCliffTime,
+    uint256 numReleasePeriods,
+    uint256 releasePeriod,
+    uint256 amountReleasedPerPeriod,
+    bool revocable,
+    address payable _beneficiary,
+    address _releaseOwner,
+    address payable _refundAddress,
+    bool subjectToLiquidityProvision,
+    uint256 initialDistributionPercentage,
+    bool _canValidate,
+    bool _canVote,
+    address registryAddress
+  ) public {
+    releaseSchedule.numReleasePeriods = numReleasePeriods;
+    releaseSchedule.amountReleasedPerPeriod = amountReleasedPerPeriod;
+    releaseSchedule.releasePeriod = releasePeriod;
+    releaseSchedule.releaseCliff = releaseStartTime.add(releaseCliffTime);
+    releaseSchedule.releaseStartTime = releaseStartTime;
+    require(releaseSchedule.numReleasePeriods >= 1, "There must be at least one releasing period");
+    require(
+      releaseSchedule.amountReleasedPerPeriod > 0,
+      "The released amount per period must be greater than zero"
+    );
+    require(
+      _beneficiary != address(0),
+      "The release schedule beneficiary cannot be the zero addresss"
+    );
+    require(registryAddress != address(0), "The registry address cannot be the zero address");
+    require(
+      releaseSchedule.releaseStartTime.add(
+        releaseSchedule.numReleasePeriods.mul(releaseSchedule.releasePeriod)
+      ) >
+        block.timestamp,
+      "Release schedule end time must be in the future"
+    );
+    require(!(revocable && _canValidate), "Revocable contracts cannot validate");
+    require(
+      initialDistributionPercentage >= 0 && initialDistributionPercentage <= 1000,
+      "Initial distribution percentage out of bounds"
+    );
+    require(
+      (revocable && _refundAddress != address(0)) || (!revocable && _refundAddress == address(0)),
+      "If contract is revocable there must be an address to refund"
+    );
+
+    setRegistry(registryAddress);
+    beneficiary = _beneficiary;
+    revocationInfo.revocable = revocable;
+    releaseOwner = _releaseOwner;
+    refundAddress = _refundAddress;
+
+    if (initialDistributionPercentage < 1000) {
+      // Cannot use `getTotalBalance()` here because the factory has not yet sent the gold.
+      uint256 totalGrant = releaseSchedule.amountReleasedPerPeriod.mul(
+        releaseSchedule.numReleasePeriods
+      );
+      // Initial percentage is expressed to 3 significant figures: [0, 1000].
+      maxDistribution = totalGrant.mul(initialDistributionPercentage).div(1000);
```

Comment:
Not sure what you mean by freeze: if you mean that we could set distribution percentage to a precision that gets rounded away, leaving less gold to be withdrawn than strictly detailed, then yes, but the response would be 1) we don't allow that much precision, so its unlikely and 2) its not a big problem, since we can just take that into account when we set the withdrawn percentages and adjust if necessary to over/under compensate